### ISSUE TITLE
MVC client sign out does not redirect to site home page

### ISSUE THREAD

**Issue by lscorcia** [IdentityServer/IdentityServer4#3830]
Hi, I'm trying to setup an ASP.net Core 3 MVC app that uses OIDC to connect to my company's SSO portal (OpenAM).

I used Visual Studio 2019 project generator to create a basic app with no authentication and then I added the OIDC client capabilities following the steps at http://docs.identityserver.io/en/latest/quickstarts/2_interactive_aspnetcore.html#creating-an-mvc-client . Logging in works great with minimal changes to the Startup class:

```
    public void ConfigureServices(IServiceCollection services)
    {
        services.AddControllersWithViews();

        // Setup Identity Server client
        JwtSecurityTokenHandler.DefaultMapInboundClaims = false;

        services.AddAuthentication(options =>
            {
                options.DefaultScheme = "Cookies";
                options.DefaultChallengeScheme = "oidc";
            })
            .AddCookie("Cookies")
            .AddOpenIdConnect("oidc", options =>
            {
                options.Authority = "https://mycompany.com/ssoservice/oauth2";
                options.RequireHttpsMetadata = false;

                options.ClientId = "openIdClient";
                options.ClientSecret = "secret";
                options.ResponseType = "code";

                options.SaveTokens = true;
            });
    }

    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
    {
        if (env.IsDevelopment())
        {
            app.UseDeveloperExceptionPage();
            IdentityModelEventSource.ShowPII = true;
        }
        else
        {
            app.UseExceptionHandler("/Home/Error");
            // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
            app.UseHsts();
        }

        app.UseHttpsRedirection();
        app.UseStaticFiles();

        app.UseRouting();

        app.UseAuthentication();
        app.UseAuthorization();

        app.UseEndpoints(endpoints =>
        {
            // endpoints.MapDefaultControllerRoute();
            endpoints.MapControllerRoute(
                name: "default",
                pattern: "{controller=Home}/{action=Index}/{id?}");
        });
    }
```

I also set up a Logout controller action:

```
    [Authorize]
    public IActionResult Logout()
    {
        return SignOut("Cookies", "oidc");
    }
```

The action actually works, i.e. when activated the cookie is deleted and the user is logged out from the SSO portal, but when the browser redirects to the /signout-callback-oidc endpoint it receives an HTTP 200 response without any content. I would have expected to have it automatically redirect to the site home page "/", which is the default value of the OpenIdConnectOptions.SignedOutRedirectUri property.

What am I missing?

**Comment by leastprivilege** [comment]
Are you saying you are asking for support on how to connect to OpenAM?

**Comment by lscorcia** [comment]
Hi, no, OpenAM integration works fine - I was just trying to supply some context. I am trying to figure out why the sample code at http://docs.identityserver.io/en/latest/quickstarts/2_interactive_aspnetcore.html#adding-sign-out does not redirect the user to the site home page, which is the intended behavior according to the last paragraph. I would have expected the SignOut action to redirect the user to the value of the OpenIdConnectOptions.SignedOutRedirectUri property.
Thanks for your help.

**Comment by leastprivilege** [comment]
So how is this related to IdentityServer?

**Comment by lscorcia** [comment]
Nevermind, I got it now. Those client features are part of the .net core libraries and not related to IdentityServer, which only provides the server side. I had not noticed that the nuget package reference to IdentityServer I had in the client app wasn't necessary, and I also misunderstood this sentence in the docs:

> This will clear the local cookie and then redirect to IdentityServer. IdentityServer will clear its cookies and then **give the user a link** to return back to the MVC application.

So totally my fault, sorry for wasting your time. On my defense, I spent most of the last days trying to work around bugs in our old version of OpenAM OIDC connector, so I guess I've been punished already.
Have a nice day!
